src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -29,8 +31,15 @@ public static partial class Certificates
 
             private static X509Certificate2Collection GetCertificateCollection(string certificateFileName)
             {
+                // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
+                // This bug results in corrupting the X509Certificate2 certificate state.
+                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
+
                 var certCollection = new X509Certificate2Collection();
                 certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+
+                m.ReleaseMutex();
```

Comment:
I thought of that but:
1.  it's not following the recommended pattern / sample from https://msdn.microsoft.com/en-us/library/system.threading.mutex(v=vs.110).aspx 
2. I believe that releasing in a `finally` is not a good idea (and yes, I don't like the `lock {}` construct either :) ): if something did fail with an exception, the state of the whole process is questionable. 
At that point, rather than releasing the mutex it may be a better idea to crash the process immediately to find out what's going on (the Assert should handle that).